tests/test_gmsd.py

```diff
@@ -16,160 +18,242 @@ def target() -> torch.Tensor:
 
-@pytest.mark.skipif(not torch.cuda.is_available(), reason='No need to run test if there is no GPU.')
-def test_multi_scale_gmsd_loss_on_gpu(prediction: torch.Tensor, target: torch.Tensor) -> None:
-    prediction = prediction.cuda()
-    target = target.cuda()
-    loss = MultiScaleGMSDLoss(chromatic=True)
-    loss(prediction, target)
+
+def test_multi_scale_gmsd_zero_for_equal_tensors(prediction: torch.Tensor, device: str):
+    target = prediction.clone()
+    measure = multi_scale_gmsd(prediction.to(device), target.to(device))
+    assert measure.abs() <= 1e-6, f'MultiScaleGMSD for equal tensors must be 0, got {measure}'
 
 
-def test_multi_scale_gmsd_loss_backward(prediction: torch.Tensor, target: torch.Tensor) -> None:
-    prediction.requires_grad_()
-    loss_value = MultiScaleGMSDLoss(chromatic=True)(prediction, target)
-    loss_value.backward()
-    assert prediction.grad is not None, LEAF_VARIABLE_ERROR_MESSAGE
+def test_multi_scale_gmsd_supports_different_data_ranges(prediction: torch.Tensor, target: torch.Tensor,
+                                                         device: str) -> None:
+    prediction_255 = (prediction * 255).type(torch.uint8)
+    target_255 = (target * 255).type(torch.uint8)
+
+    measure = multi_scale_gmsd(prediction.to(device), target.to(device))
+    measure_255 = multi_scale_gmsd(prediction_255.to(device), target_255.to(device), data_range=255)
+    diff = torch.abs(measure_255 - measure)
+    assert diff <= 1e-4, f'Result for same tensor with different data_range should be the same, got {diff}'
+
+
+def test_multi_scale_gmsd_supports_greyscale_tensors(device: str):
+    target = torch.ones(2, 1, 128, 128)
+    prediction = torch.zeros(2, 1, 128, 128)
+    multi_scale_gmsd(prediction.to(device), target.to(device))
+
+
+def test_multi_scale_gmsd_fails_for_greyscale_tensors_chromatic_flag(device: str):
+    target = torch.ones(2, 1, 128, 128)
+    prediction = torch.zeros(2, 1, 128, 128)
+    with pytest.raises(AssertionError):
+        multi_scale_gmsd(prediction.to(device), target.to(device), chromatic=True)
+
+
+def test_multi_scale_gmsd_supports_custom_weights(prediction: torch.Tensor, target: torch.Tensor, device: str):
+    multi_scale_gmsd(prediction.to(device), target.to(device), scale_weights=[3., 4., 2., 1., 2.])
+    multi_scale_gmsd(prediction.to(device), target.to(device), scale_weights=torch.tensor([3., 4., 2., 1., 2.]))
 
 
-@pytest.mark.skipif(not torch.cuda.is_available(), reason='No need to run test if there is no GPU.')
-def test_multi_scale_gmsd_loss_backward_on_gpu(prediction: torch.Tensor, target: torch.Tensor) -> None:
+def test_multi_scale_gmsd_raise_exception_for_small_images(device: str):
+    target = torch.ones(3, 1, 32, 32)
+    prediction = torch.zeros(3, 1, 32, 32)
+    with pytest.raises(ValueError):
+        multi_scale_gmsd(prediction.to(device), target.to(device), scale_weights=[3., 4., 2., 1., 1.])
```

Comment:
Since you are already testing some resolutions that are fine and some resolutions that are too small, could you please estimate the threshold (may be just experimentally?) and put it somewhere in the docsting for the metric and also in the README. It will be super valuable for a user to now this value beforehand